Lib/typing.py

```diff
@@ -1959,11 +1959,13 @@ def no_type_check(arg):
     This mutates the function(s) or class(es) in place.
     """
     if isinstance(arg, type):
-        arg_attrs = arg.__dict__.copy()
-        for attr, val in arg.__dict__.items():
-            if val in arg.__bases__ + (arg,):
-                arg_attrs.pop(attr)
-        for obj in arg_attrs.values():
+        for obj in arg.__dict__.values():
+            if (hasattr(obj, '__qualname__')
```

Comment:
This isn't enough if the module is also different. (I think Alex brought that up already.)